### PR TITLE
Rename ITrace to ITraceSpan

### LINKED ISSUE #8: Rename ITrace to ISpan
The interface that represents a span is named `ITrace`, but a trace is a graph of spans. Getting the terminology straight was hard enough when I was first looking at OpenTracing; we shouldn't make it harder by not using the most obvious names.

### PR DESCRIPTION
**Pull request by bgrainger**
Rename `ITrace` to `ITraceSpan`, `ITraceProvider` to `ITraceSpanProvider`, `TraceKind` to `TraceSpanKind`, etc.

Fixes #8.

### CODE REVIEW

**Comment by bgrainger** [comment]
Some methods or types (e.g., `CreateSpan`, `SpanTagNames`, `StartActionSpan`) use just "Span", not "TraceSpan", where usage didn't seem to be ambiguous. These could be "TraceSpan" everywhere (for consistency) if someone felt strongly about it; I thought the current blend was good.

Actually, I think I'll rename `ITraceSpanProvider.CurrentTraceSpan` to just `.CurrentSpan`.

File: src/Faithlife.Tracing.Http/HttpTracingUtility.cs [hunk 1]
```diff
@@ -7,65 +7,65 @@ namespace Faithlife.Tracing.Http
...
-				var requestTrace = SetHttpHeadersAndCreateTrace(currentTrace, request.RequestUri, (k, v) => request.Headers.Add(k, v));
-				if (requestTrace != null)
+				var currentSpan = m_traceSpanProvider?.CurrentSpan;
+				var requestSpan = SetHttpHeadersAndCreateSpan(currentSpan, request.RequestUri, (k, v) => request.Headers.Add(k, v));
+				if (requestSpan != null)
 				{
-					requestTrace.SetTag(TraceTagNames.HttpMethod, request.Method.ToString());
-					request.Properties[c_traceKey] = requestTrace;
+					requestSpan.SetTag(SpanTagNames.HttpMethod, request.Method.ToString());
+					request.Properties[c_traceKey] = requestSpan;
```
Comment (ddunkin) [inline src/Faithlife.Tracing.Http/HttpTracingUtility.cs]: `c_traceKey` should be renamed (and its value changed).

File: src/Faithlife.Tracing.AspNet/RequestActionTraceSpanProvider.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+namespace Faithlife.Tracing.AspNet
+{
+	/// <summary>
+	/// An <see cref="ITraceSpanProvider"/> that has a "stack" of two trace spans: the request
+	/// and the controller action method; it returns the deepest span in the stack as
+	/// the current one.
+	/// </summary>
+	internal sealed class RequestActionTraceSpanProvider : ITraceSpanProvider
+	{
+		public ITraceSpan CurrentTraceSpan => m_actionSpan ?? m_requestSpan;
```
Comment (ddunkin) [inline src/Faithlife.Tracing.AspNet/RequestActionTraceSpanProvider.cs]: `CurrentSpan`?

**Review by ddunkin — COMMENTED** [review]
"TraceSpan" *is* verbose, and there is some inconsistency with some code using "TraceSpan" and some using "Span". Is confusion with `Span<T>` really going to be an issue?

**Review by ejball — APPROVED** [review]
> I thought the current blend was good

I agree; only top-level types need to be disambiguated, not members.

**Pull request merged**